staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1086,7 +1093,7 @@ func (e *Store) DeleteCollection(ctx context.Context, options *metav1.DeleteOpti
 					errs <- err
 					return
 				}
-				if _, _, err := e.Delete(ctx, accessor.GetName(), options); err != nil && !kubeerr.IsNotFound(err) {
+				if _, _, err := e.Delete(ctx, accessor.GetName(), deleteValidation, options); err != nil && !kubeerr.IsNotFound(err) {
```

Comment:
We call admission for each element in the collection?

Reply:
> Shouldn't we force the check for every item of it?

If we have [remote admission controllers](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) set, this will not be that good.

e.g. I delete a collection of 1000 elements, than the api-server will call the remote admission controller 1000 times, this will be a flood I think.

Reply:
@CaoShuFeng Thanks, these changes reverted.